lib/net.js

```diff
@@ -218,6 +218,49 @@ function normalizeArgs(args) {
   return arr;
 }
 
+// Normalize the options object passed to Server.prototype.listen
+function normalizeHostOption(options) {
+  let hostname, port;
+
+  try {
+    // eslint-disable-next-line
+    const parsedHost = new URL(`http://${options.host}`);
+    hostname = parsedHost.hostname;
+    port = parsedHost.port;
+  } catch {
+    // Some IPv6 addresses are not parsed correctly
+    if (isIPv6(options.host)) {
+      hostname = options.host;
+      port = '';
+    } else {
+      // eslint-disable-next-line
+      throw new TypeError('invalid `host` value');
```

Comment:
Rather than throwing TypeErrors directly, can you use the infrastructure from `lib/internal/errors.js`?

Reply:
Absolutely. As mentioned in the initial post, if this is an interesting change this is one of the things that will be addressed.